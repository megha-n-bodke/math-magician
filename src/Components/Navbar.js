import React from 'react';

class Navbar extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    return (
      <div className="navbar">
        <h2>Math Magician</h2>
        <ul className="navLinks">
          <li>Home</li>
          <li>Calculator</li>
          <li>Quote</li>
        </ul>
      </div>
    );
  }
}

export default Navbar;
