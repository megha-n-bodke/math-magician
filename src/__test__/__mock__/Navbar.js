import React from 'react';
import { BrowserRouter, Link } from 'react-router-dom';

const Navbar = () => (
  <BrowserRouter>
    <div className="navbar">
      <h2>Math Magician</h2>
      <ul className="navLinks">
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="Calculator">Calculator</Link>
        </li>
        <li>
          <Link to="Quotes">Quotes</Link>
        </li>
      </ul>
    </div>
  </BrowserRouter>
);

export default Navbar;
