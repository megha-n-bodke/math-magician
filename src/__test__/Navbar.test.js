import React from 'react';
import renderer from 'react-test-renderer';
import Navbar from './__mock__/Navbar';

// import { RouterWrapper } from "../path/to/RouterWrapper";

test('Navbar snapshot testing', () => {
  const navbar = renderer.create(<Navbar />);
  const navObject = navbar.toJSON();
  expect(navObject).toMatchSnapshot();
});
