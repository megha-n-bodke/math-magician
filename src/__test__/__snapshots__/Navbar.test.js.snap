// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot testing 1`] = `
<div
  className="navbar"
>
  <h2>
    Math Magician
  </h2>
  <ul
    className="navLinks"
  >
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        href="/Calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        href="/Quotes"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</div>
`;
